### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by JJaug**

### CODE REVIEW

File: AnagramSolver.Tests/WordRepositoryTests.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
+﻿using AnagramSolver.BusinessLogic.Classes;
+using NUnit.Framework;
+using System;
+
+namespace AnagramSolver.Tests
+{
+    [TestFixture]
+    public class WordRepositoryTests
+    {
+        private WordRepository _wordRepository;
```
Comment (IgnasJank) [inline AnagramSolver.Tests/WordRepositoryTests.cs:11]: Make this private variable readonly or const. 

Read a little bit about the differences.

File: AnagramSolver.Tests/AnagramSolverTests.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
+﻿using NUnit.Framework;
+
+namespace AnagramSolver.Tests
+{
+    [TestFixture]
+    class AnagramSolverTests
+    {
+        private BusinessLogic.Classes.AnagramSolver _anagramSolver;
+        public const string _filePath = "C:\\Users\\jonas.jaugelis\\source\\repos\\AnagramSolver\\zodynas.txt";
```
Comment (IgnasJank) [inline AnagramSolver.Tests/AnagramSolverTests.cs:9]: Read a bit more about absolute and relative paths.

File: AnagramSolver.BusinessLogic/Classes/AnagramSolver.cs [hunk 3]
```diff
@@ -8,49 +8,108 @@ namespace AnagramSolver.BusinessLogic.Classes
...
     public class AnagramSolver : IAnagramSolver
     {
-        WordRepository wordRepository = new WordRepository();
+        IWordRepository wordRepository = new WordRepository();
         public List<Anagram> GetAnagrams(string command, int minLength, string filePath)
         {
             var allWords = wordRepository.GetAllWords(filePath, minLength);
 
-            // var exists = allWords.FirstOrDefault(p => p.Equals("sula"));
+            List<Anagram> newList = new List<Anagram>();
```
Comment (KristupasLunskas) [inline AnagramSolver.BusinessLogic/Classes/AnagramSolver.cs]: var could be used here

File: AnagramSolver.BusinessLogic/Classes/WordRepository.cs [hunk 4]
```diff
@@ -8,23 +8,36 @@ namespace AnagramSolver.BusinessLogic.Classes
...
-                    word.Word = itemsInLine[2];
-                    newList.Add(word);
+                    if (itemsInLine[2].Length >= minLength)   
+                    {
+                        newList.Add(itemsInLine[2]);
+                    }
                 }
+                catch
+                {
+                    return null; ;
```
Comment (KristupasLunskas) [inline AnagramSolver.BusinessLogic/Classes/WordRepository.cs]: double ; ;

File: AnagramSolver.Models/Models/Anagram.cs [hunk 5]
```diff
@@ -2,7 +2,6 @@
 {
     public class Anagram
```
Comment (KristupasLunskas) [inline AnagramSolver.Models/Models/Anagram.cs]: At the moment this class only contains one string, which means this class can be removed and the code can be refactored (after tests cover the area you refactor)

File: AnagramSolver.Models/Models/Words.cs [hunk 6]
```diff
@@ -1,14 +1,8 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-
-namespace AnagramSolver.Models.Models
+﻿namespace AnagramSolver.Models.Models
 {
     public class Words
```
Comment (KristupasLunskas) [inline AnagramSolver.Models/Models/Words.cs]: At the moment this class only contains one string, which means this class can be removed and the code can be refactored (after tests cover the area you refactor)

File: AnagramSolver.Tests/AnagramSolverTests.cs [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        [TestCase("rimti", "mirti")]
+        public void Should_ReturnOneAnagram_When_GivenOneWord(string command, string anagram)
+        {
+            var allAnagrams = _anagramSolver.GetAnagrams(command, 4, @_filePath);
+
+            Assert.That(allAnagrams[0].Word, Is.EqualTo(anagram));
+        }
+
+        [Test]
+        [TestCase("balos")]
```
Comment (KristupasLunskas) [inline AnagramSolver.Tests/AnagramSolverTests.cs]: Discussed in meeting about TestCase usage and good practice.
More information about AAA in Unit testing https://www.c-sharpcorner.com/UploadFile/dacca2/fundamental-of-unit-testing-understand-aaa-in-unit-testing/

File: AnagramSolver.Tests/AnagramSolverTests.cs [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        public void Should_ReturnAllAnagrams_When_GivenOneWord(string command)
+        {
+            var allAnagrams = _anagramSolver.GetAnagrams(command, 4, @_filePath);
+
+            Assert.That(allAnagrams.Count, Is.GreaterThan(1));
```
Comment (KristupasLunskas) [inline AnagramSolver.Tests/AnagramSolverTests.cs:35]: Discussed in meeting about different Assert's methods. They are indifferent, and next syntax is very welcoming. However it's good to know classic syntax as well. In this example `Assert.Greater(actualcount, 1);` would be considered classic syntax. In some cases this is easier to understand what is actual and what is expected value.

File: AnagramSolver.Tests/WordRepositoryTests.cs [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+
+        [Test]
+        public void Should_ReturnListOfWords_When_GivenFileOfWords()
+        {
+            var allWords = _wordRepository.GetAllWords(_filePath, 4);
+            Assert.That(allWords, !Is.Null);
+        }
+        [Test]
+        [TestCase("C:\\Users\\jonas.jaugelis\\source\\repos\\AnagramSolver\\zodynas2.txt")]
```
Comment (KristupasLunskas) [inline AnagramSolver.Tests/WordRepositoryTests.cs]: Discussed about TestCase in a meeting. Consider adding a variable path in Assert section of test.

File: AnagramSolver.BusinessLogic/Classes/AnagramSolver.cs (hunk 3 above)
Comment (IgnasJank) [reply AnagramSolver.BusinessLogic/Classes/AnagramSolver.cs]: Also a possibility to write like that in .NET 5:
`List<Anagram> newList = new();`
However, it is a new and not popular approach yet.

Comment (JJaug) [reply AnagramSolver.BusinessLogic/Classes/AnagramSolver.cs]: sorry about that. I think I will stick with var, more understendable

File: AnagramSolver.Tests/WordRepositoryTests.cs (hunk 1 above)
Comment (JJaug) [reply AnagramSolver.Tests/WordRepositoryTests.cs:11]: If made it const or readonly i get errors all over the place

**Pull request merged**